Kernel/MLMatcher.cpp

```diff
@@ -506,6 +501,8 @@ void MLMatcher::Impl::initMatchingData(Literal** baseLits0, unsigned baseLen, Cl
       zeroAlts++;
       singleAlts++;
     } else if(currAltCnt==1 && !(resolvedLit && resolvedLit->couldBeInstanceOf(s_baseLits[i], true)) ) {
+      //TODO the above requires updating to polymorphism
+      //for now, a minor source of incompleteness
```

Comment:
I have forgotten... Do we have the mechanism in place, not to report SAT on polymorphic inputs which we finitely saturate?

Reply:
No we don't. I think that the current thing to do would be to fix the above, which hopefully I can do with Jakob's help. If not, I will edit the reporting of SAT.

Reply:
Looking through the code, It seems that I already dealt with the above issue.

`couldBeInstanceOf` makes a call to `Literal::headersMatch` which in `master` contains a syntactic equality sort check making it unsuitable. However, I have removed this check from the polymorphic version and pushed it into the places in the code that call `Literal::headersMatch`. 

If you agree with this, I think the comment can be deleted and this issue resolved.

Reply:
Sure, go for it!